### PR TITLE
check `badger.ErrKeyNotFound` error returned by `txn.Get`.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

File: tikv/mvcc.go [hunk 1]
```diff
@@ -265,9 +265,12 @@ func (store *MVCCStore) Commit(req *requestCtx, keys [][]byte, startTS, commitTS
 func (store *MVCCStore) handleLockNotFound(reqCtx *requestCtx, key []byte, startTS, commitTS uint64) error {
 	txn := reqCtx.getDBReader().txn
 	item, err := txn.Get(key)
-	if err != nil {
+	if err != nil && err != badger.ErrKeyNotFound {
 		return errors.Trace(err)
 	}
+	if item == nil {
```
Comment (ngaut) [inline tikv/mvcc.go]: Check if err if badger.ErrKeyNotFound  should be more clear.

File: tikv/mvcc.go [hunk 2]
```diff
@@ -339,7 +342,7 @@ func (store *MVCCStore) rollbackKey(req *requestCtx, batch *writeBatch, key []by
 	}
 	reader := req.getDBReader()
 	item, err := reader.txn.Get(key)
-	if err != nil {
+	if err != nil && err != badger.ErrKeyNotFound {
```
Comment (ngaut) [inline tikv/mvcc.go]: What happend if err is badger.ErrKeyNotFound?

Comment (coocood) [reply tikv/mvcc.go]: Then `item` would be `nil`

File: tikv/mvcc.go (hunk 1 above)
Comment (coocood) [reply tikv/mvcc.go]: I think `badger.ErrKeyNotFound` is redundant, check if the returned object is nil is a more commonly used pattern.

**Pull request merged**